plugins/inputs/mesos/mesos.go

```diff
@@ -593,11 +672,119 @@ func (m *Mesos) gatherMainMetrics(u *url.URL, role Role, acc telegraf.Accumulato
 		}
 	}
 
+	var includeTaggedFields bool
+	for _, col := range m.MasterCols {
+		if col == "framework_offers" || col == "allocator" {
+			includeTaggedFields = true
+			break
+		}
+	}
+
+	taggedFields := map[string][]TaggedField{}
+	extraTags := map[string]map[string]string{}
+
+	if includeTaggedFields {
+		for metricName, val := range jf.Fields {
```

Comment:
I do not understand the logic here. I understand the code but I do not know why we do it.

Reply:
Sorry for not explaining the background of the ticket better - For all of the framework offers metrics and some of the allocator metrics, there are variables in the path that we need to pull out into tags and exclude from the metric name. e.g. all the framework offer metrics are prefixed with `master/frameworks/<ENCODED_FRAMEWORK_NAME>/<FRAMEWORK_ID>/...` - we want the resulting metric name to look like `master/frameworks/...`so we need to pull out the `<ENCODED_FRAMEWORK_NAME>` into a `framework_name: <ENCODED_FRAMEWORK_NAME>` tag; and we've decided it wasn't necessary to include a `<FRAMEWORK_ID>` tag. All previously supported metrics did not involve variables, thus this newly introduced logic to handle these metric types.